### ISSUE TITLE
Hal Browser demo doesn't work

### ISSUE THREAD

**Issue by tomq42** [mikekelly/hal-browser#77]
Attempting to use 
http://haltalk.herokuapp.com/explorer/browser.html
doesn't work.

If I just use GET /users, I get an internal server error.
Attempting to POST /users to create a new user leads to a 404.

**Comment by mikekelly** [comment]
Yep, it needs some love. A heroku upgrade triggered a bunch of library incompatibilities. I did try and fix the dependencies but it was taking too long with no results, I intend to rewrite it if/when I get time.

**Comment by tomq42** [comment]
OK, no problem.
Thing is there's not a lot of doc about HAL. Specifically how collections are
represented, which is slightly unusual IMO. So I was trying to find some
examples. I eventually found some examples elsewhere (apigility I think).
 
Thanks.

**Comment by dylanbeattie** [comment]
@tomq42 - have you seen the docs at http://phlyrestfully.readthedocs.org/en/latest/halprimer.html#collections ? Some very useful examples in there of how to model collections in HAL.

**Comment by tomq42** [comment]
Yes, but it was so weird that I don't think I understood it originally. But I
think I understand the example a bit better now.

**Comment by linde12** [comment]
Would be nice to see the demo page fixed indeed!

**Comment by hnilsen** [comment]
Just dropping in to give my support :-)

**Comment by nerdyglasses** [comment]
Hey yeah. I'd like to help too if there is something I can do.

**Comment by mikekelly** [comment]
Hi all. So the rails app is a write-off, it uses a very old database driver. It was something I hacked together in a couple of hours anyway.

I think the best course of action is for someone to create a a new implementation of the hal-talk API. I don't have the bandwidth to do this at the moment. If someone wants to take this on and put it on a public github repo, I'll setup the heroku app so that it auto-deploys from it.

**Comment by mikekelly** [comment]
I can dump the existing data, and extract the link relation docs... so if someone does want to take this one let me know and I'll do that

**Comment by mhogerheijde** [comment]
So we're talking about having a small app that implements a few API calls backed by some sort of persistence layer, right? The front-end itself is just (statically served) Javascript, right?

What about the API documentation? That's served (statically) by the backend as well right?

Do you have an opinion on the Tech-stack that will be used?

**Comment by mikekelly** [comment]
Not really, preferably something simple and lightweight. Ruby or node.js or similar. It doesn't matter too much to be honest.

I think basically the API just needs some use cases that demonstrates key concepts in a HAL API, ie: traversing links (read, write, delete), embedding resources, etc

I picked "talk" just because it's a relatively simple read/write domain

**Comment by mhogerheijde** [comment]
All right, coming weekend is a busy one for me, so I'll try to make a start next week/weekend. I'll keep you posted.
